Sources/HipMobileUI/ViewModels/Pages/UserRatingPageViewModel.cs

```diff
@@ -0,0 +1,369 @@
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//       http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+using Acr.UserDialogs;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Managers;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Helpers;
+using System;
+using System.Windows.Input;
+using Xamarin.Forms;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Resources;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common.Contracts;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models.ModelClasses;
+using System.IO;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.DtoToModelConverters;
+using System.Collections.Generic;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Pages
+{
+    public class UserRatingPageViewModel : NavigationViewModel
+    {
+
+        #region Variabels
+        private ImageSource image;
+        private Exhibit exhibit;
+        private string headline;
+        private ICommand sendRatingCommand;
+        private ICommand selectStarCommand;
+
+        private string star1;
+        private string star2;
+        private string star3;
+        private string star4;
+        private string star5;
+
+        private GridLength star1Bar;
+        private GridLength star2Bar;
+        private GridLength star3Bar;
+        private GridLength star4Bar;
+        private GridLength star5Bar;
+
+        private string star1BarCount;
+        private string star2BarCount;
+        private string star3BarCount;
+        private string star4BarCount;
+        private string star5BarCount;
+
+        private string ratingAverage;
+        private string ratingStars;
+        private string ratingCount;
+
+        private int rating = 0;
+        #endregion
+
+
+        public UserRatingPageViewModel(Exhibit exhibit)
+        {
+            Exhibit = exhibit;
+            Headline = exhibit.Name;
+
+            var imageData = exhibit.Image.GetDataBlocking();
+            Image = imageData != null ? ImageSource.FromStream(() => new MemoryStream(imageData)) : ImageSource.FromStream(() => new MemoryStream(BackupData.BackupImageData));
+
+            SetUserRatingUi();
+            SendRatingCommand = new Command(SendUserRating);
+            SelectStarCommand = new Command(OnSelectStar);
+
+        }
+        private async void SetUserRatingUi()
```

Comment:
There should be a blank line before this. Also again, is `Set[...]` the best name for this?